File: frontend/src/types/ProblemDTO.ts
import { TestCaseDTO } from "./TestCaseDTO";
import { TopicDTO } from "./TopicDTO";

export interface ProblemDTO {
    id: number | null;
    title: string;
    author: string;
    description: string;
    constraints: string;
    difficulty: "easy" | "medium" | "hard";
    officialSolution: string | null;
    image: string | null;
    tests: TestCaseDTO[];
    topics: TopicDTO[];
}

File: backend/src/main/java/org/example/backend/dto/ProblemDTO.java
package org.example.backend.dto;

import java.util.List;

public record ProblemDTO(
        Long id,
        String title,
        String author,
        String description,
        String constraints,
        String difficulty,
        String officialSolution,
        String image,
        List<TestCaseDTO> tests,
        List<TopicDTO> topics) {
}
